editor/src/core/shared/multiplayer-hooks.tsx

```diff
@@ -37,3 +55,180 @@ export function useMyUserId(): string | null {
   )
   return myUserId
 }
+
+export function useIsBeingFollowed() {
+  const mode = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.mode,
+    'useIsBeingFollowed mode',
+  )
+
+  const isBeingFollowed = React.useCallback(
+    (connectionId: number) => {
+      return isFollowMode(mode) && mode.connectionId === connectionId
+    },
+    [mode],
+  )
+
+  return isBeingFollowed
+}
+
+export interface SortableUser {
+  connectionId: number
+}
+
+export function useSortMultiplayerUsers() {
+  const isBeingFollowed = useIsBeingFollowed()
+  return React.useCallback(
+    (a: SortableUser, b: SortableUser) => {
+      if (isBeingFollowed(a.connectionId)) {
+        return -1
+      }
+      if (isBeingFollowed(b.connectionId)) {
+        return 1
+      }
+      if (a.connectionId < b.connectionId) {
+        return -1
+      }
+      if (b.connectionId < a.connectionId) {
+        return 1
+      }
+      return 0
+    },
+    [isBeingFollowed],
+  )
+}
+
+export function useMyConnections(): ConnectionInfo[] {
+  const myUserId = useMyUserId()
+  const conns = useStorage((store) => store.connections)
+  if (myUserId == null) {
+    return []
+  }
+  return conns[myUserId] ?? []
+}
+
+/**
+ * Store the current user's connection in the room storage.
+ */
+export function useStoreConnection() {
+  const loginState = useEditorState(
+    Substores.userState,
+    (store) => store.userState.loginState,
+    'useAddConnection loginState',
+  )
+
+  const storeConnection = useMutation(
+    ({ storage, self }) => {
+      if (!isLoggedIn(loginState)) {
+        return
+      }
+
+      const selfConns: ConnectionInfo[] = storage.get('connections').get(self.id) ?? []
+
+      // if the current connection is not stored yet...
+      if (!selfConns.some((c) => c.id === self.connectionId)) {
+        // ...store it...
+        const now = Date.now()
+        selfConns.push({
+          id: self.connectionId,
+          startedAt: now,
+          lastSeen: now,
+        })
+        storage.get('connections').update({ [self.id]: selfConns })
```

Comment:
I surmise that there's the possibility that something else changes this after the lookup and then we just overwrite it, not sure if we actually care, but it seems worth mentioning.

Reply:
indeed and it would be great to lock the thing... I did this to avoid having `N` update calls, but maybe they would be less error prone 🤔

Reply:
I don't _think_ it's necessary, because we're just gonna get variations on a theme of the values I believe. They'd just get slightly different times.